components/Chat/Chat.tsx

```diff
@@ -241,8 +241,31 @@ const getListArray = (
   return reverseArray;
 };
 
+const useAnimatedListView = (
+  conversation: XmtpConversationWithUpdate | undefined
+) => {
+  // The first message was really buggy on iOS & Android and this is due to FlashList
+  // so we keep FlatList for new convos and switch to FlashList for bigger convos
+  // that need great perf.
+  return useMemo(() => {
+    const isConversationNotPending = conversation && !conversation.pending;
+    return isConversationNotPending && Platform.OS !== "web"
+      ? ReanimatedFlashList
+      : ReanimatedFlatList;
+  }, [conversation]);
+};
+
+const useIsShowingPlaceholder = (
+  listArray: MessageToDisplay[],
+  isBlockedPeer: boolean,
+  conversation: XmtpConversationWithUpdate | undefined
+): boolean => {
+  return listArray.length === 0 || isBlockedPeer || !conversation;
+};
```

Comment:
I like that when there's more than 1 argument we use object arguments instead. But my own opinion. Not sure if we have some specifics docs around that.

Reply:
For a hook like that I would do this. (again just my own opinion lol)

```
const useIsShowingPlaceholder = (args: {
  listArray: MessageToDisplay[];
  isBlockedPeer: boolean;
  hasConversation: boolean;
}): boolean => {
  return args.listArray.length === 0 || args.isBlockedPeer || ! args.hasConversation;
};
```

Reply:
Hey @thierryskoda

Thanks for your feedback. As you suggested, I've refactored `useIsShowingPlaceholder` and `useRenderItem` to use the object parameter pattern. I did direct destructuring in the function parameters rather than using 'args'.

Cheers!